game/plugins/src/main/kotlin/gg/rsmod/plugins/content/mechanics/prayer/prayers.plugin.kts

```diff
@@ -1,5 +1,30 @@
 package gg.rsmod.plugins.content.mechanics.prayer
 
+/**
+ * Checks if Retribution is active and deals
+ * ((prayer lvl)/4) damage to nearby enemies
+ */
+on_player_pre_death {
+    if (Prayers.isActive(player, Prayer.RETRIBUTION)) {
+        val targets = player.getNeighbourPawns()
+        for (target in targets) {
+            if (!target.isDead()) {
+                if (target is Player && target.lock.canBeAttacked()) {
+                    // TODO Only deal damage in pvp situations
+                    player.graphic(437)
+                    target.hit(damage = (player.getSkills().getCurrentLevel(Skills.PRAYER) / 4))
+                } else if (target is Npc) {
+                    if (target.isSpawned() && target.def.isAttackable() && target.lock.canBeAttacked()) {
+                        player.graphic(437)
```

Comment:
Both statements for players and npc's are identical, perhaps do something at the start of the loop like `target is not NPC || target is not Player => continue` and execute the code beneath there.